var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -155,6 +155,31 @@ def determine_toolset(self, spec):
         # fallback to gcc if no toolset found
         return 'gcc'
 
+    def bjam_python_line(self, spec):
+        from glob import glob
+        from os.path import dirname, splitext
+        pydir = 'python%s.%s*' % spec['python'].version.version[:2]
+        incs = join_path(spec['python'].prefix.include, pydir, "pyconfig.h")
+        incs = glob(incs)
+        incs = " ".join([dirname(u) for u in incs])
+
+        pylib = 'libpython%s.%s*' % spec['python'].version.version[:2]
+        alllibs = join_path(spec['python'].prefix.lib, pylib)
+        if sys.platform == 'darwin':
```

Comment:
just use `dso_suffix`, which is `dylib` or `so` based on platform. Then the whole `if` could go.